### DISCUSSION TITLE
Dockerize project

### DISCUSSION THREAD

**Discussion by coskundeniz89** [jjdejong/phpip#128] [Ideas]
I need to use your project so could you dockerize your project?

**Comment by jjdejong** [comment]
We only develop things as we need them ourselves, and contribute the developement freely to this repository - why wouldn't you contribute such a dockerization? Others may find it very useful!

**Comment by jjdejong** [comment]
This project is not optimal for containerizing, because updates would be complex to implement by the user. Each code base update could be applied by installing a new docker image, but database table updates would need to be applied manually on top of installing a new docker image. It is way simpler to do a `git pull` and then a `php artisan migrate` to apply an update.

The `php artisan migrate` script could be run automatically when installing a new docker image, but then it's hard to check for eventual script errors and even harder to fix them.

**Comment by papoteur-mga** [comment]
I gave a first try for building a docker image.
Here are [instructions](https://github.com/jjdejong/phpip/wiki/Docker-installation)
I'm not an expert of docker, thus this can be probably improved.
